src/init/mod.rs

```diff
@@ -121,68 +121,60 @@ pub fn init_stub(shell_name: &str) -> io::Result<()> {
+                fi
+            }}
+            __main
+            unset -f __main
+            "#,
+            starship.sprint_posix()?
+        ),
+        "zsh" => print!(
+            r#"source <("{}" init zsh --print-full-init)"#,
+            starship.sprint_posix()?
+        ),
+        "fish" => print!(
             // Fish does process substitution with pipes and psub instead of bash syntax
-            let script = format!(
-                "source (\"{}\" init fish --print-full-init | psub)",
-                starship.sprint_posix()?
-            );
-            Some(script)
-        }
-        "powershell" => {
-            // Explanation of syntax:
-            // &: Explicitly tells powershell to execute path with starship executable.
-            //
-            // @: multi-line stdout is returned as an array, but a single line or no lines
-            //    are returned as-is. @ ensures it's always an array.
-            //
-            // -join "`n": Joins the stdout array together as a string with newlines.
-            //             Powershell escapes with ` instead of \ thus `n translates to a newline.
-            let script = format!(
-                "Invoke-Expression (@(&\"{}\" init powershell --print-full-init) -join \"`n\")",
-                starship.sprint()?
-            );
-            Some(script)
-        }
-        "ion" => {
-            let script = format!("eval $({} init ion --print-full-init)", starship.sprint()?);
-            Some(script)
-        }
-        "elvish" => {
-            let script = format!(
-                "eval (\"{}\" init elvish --print-full-init | slurp)",
-                starship.sprint_posix()?
-            );
-            Some(script)
-        }
-        "tcsh" => {
-            let script = format!(
-                r#"eval "`("{}" init tcsh --print-full-init)`""#,
-                starship.sprint_posix()?
-            );
-            Some(script)
-        }
+            r#"source ("{}" init fish --print-full-init | psub)"#,
+            starship.sprint_posix()?
+        ),
+        "powershell" => print!(
+            /*
+             * Explanation of syntax:
+             *
+             *      iex: alias of Invoke-Expression
+             *
+             *      &: Explicitly tells PowerShell to execute path with Starship
+             *         executable.
+             *
+             *      @: multi-line stdout is returned as an array, but a single
+             *         line or no lines are returned as-is. @ ensures it's always
+             *         an array.
+             *
+             *      -join "`n": Joins the stdout array together as a string with
+             *                  newlines. PowerShell escapes with ` instead of \
+             *                  thus `n translates to a newline.
+             */
+            r#"iex (@(&"{}" init powershell --print-full-init) -join "`n")"#,
```

Comment:
The PowerShell linter recommends against using shorthands in PowerShell scripts.

But for simplification using `Out-String` over ``-join "`n"`` might make sense.

Reply:
I've just swapped `Invoke-Expression` with `iex` here because I got used to seeing it around inside various installation instructions. Everything else is inherited.
So you are suggesting going back to `Invoke-Expression`, and replacing ``-join "`n"`` with `Out-String`? Whatever you like, just please write it down cause I'm doing blind coding here. I've never used posh longer than few minutes and don't have an installation at hand for testing :upside_down_face:

Reply:
This will work:

```suggestion
            r#"Invoke-Expression (&"{}" init powershell --print-full-init | | Out-String)"#,
```

Reply:
Now it makes sense to completely lose the above comment? Names are pretty descriptive and I would guess that `&` operator is a well-known thing in the PowerShell world?

Reply:
I think it's fine to remove the comment.

Reply:
Done, thanks for your help! :+1: